services/web/server/tests/unit/with_dbs/01/test_director_v2.py

```diff
@@ -117,5 +137,72 @@ async def test_get_computation_task(
     assert task_out.state == RunningState.NOT_STARTED
 
 
-async def test_delete_pipeline(client, user_id: PositiveInt, project_id: UUID):
+async def test_delete_pipeline(
+    mocked_director_v2, client, user_id: UserID, project_id: ProjectID
+):
     await director_v2_api.delete_pipeline(client.app, user_id, project_id)
+
+
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+@given(cluster_create=st.builds(ClusterCreate))
+async def test_create_cluster(
+    mocked_director_v2, client, user_id: UserID, cluster_create
+):
+    created_cluster = await director_v2_api.create_cluster(
+        client.app, user_id=user_id, new_cluster=cluster_create
+    )
+    assert created_cluster is not None
+    assert isinstance(created_cluster, dict)
+    assert "id" in created_cluster
+
+
+async def test_list_clusters(mocked_director_v2, client, user_id: UserID):
+    list_of_clusters = await director_v2_api.list_clusters(client.app, user_id=user_id)
+    assert isinstance(list_of_clusters, list)
+    assert len(list_of_clusters) > 0
+
+
+async def test_get_cluster(
+    mocked_director_v2, client, user_id: UserID, cluster_id: ClusterID
+):
+    cluster = await director_v2_api.get_cluster(
+        client.app, user_id=user_id, cluster_id=cluster_id
+    )
+    assert isinstance(cluster, dict)
+    assert cluster["id"] == cluster_id
+
+
+async def test_get_cluster_details(
+    mocked_director_v2, client, user_id: UserID, cluster_id: ClusterID
+):
+    cluster_details = await director_v2_api.get_cluster_details(
+        client.app, user_id=user_id, cluster_id=cluster_id
+    )
+    assert isinstance(cluster_details, dict)
+
+
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+@given(cluster_patch=st.from_type(ClusterPatch))
+async def test_update_cluster(
```

Comment:
Question on hypothesis. How does this work? Is there a way to configure how many times this is ran and also if it detects a failure does it report it, or only if it goes over a certain threshold?

Reply:
google is your friend: [https://hypothesis.readthedocs.io/en/latest/](https://hypothesis.readthedocs.io/en/latest/)

Reply:
and yes all this is possible